### PR TITLE
DT-1734 Consolidate tests to prepare for splitting out e2e tests

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt [hunk 1]
```diff
@@ -85,25 +82,4 @@ class JmsConfig {
       .withEndpointConfiguration(AwsClientBuilder.EndpointConfiguration(serviceEndpoint, region))
       .withCredentials(AWSStaticCredentialsProvider(AnonymousAWSCredentials()))
       .build()
-
-  @Bean
-  @ConditionalOnProperty(name = ["sqs.provider"], havingValue = "localstack")
-  @Suppress("SpringJavaInjectionPointsAutowiringInspection")
-  open fun queueUrl(
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt:92]: Removed queue URL and DLQ URL beans as they can be derived on the fly.

File: src/testIntegration/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/HouseKeepingIntegrationTest.kt [hunk 2]
```diff
@@ -6,30 +6,19 @@ import org.assertj.core.api.Assertions.within
...
-  @Inject
-  private lateinit var messageRepository: MessageRepository
+class HouseKeepingIntegrationTest : QueueListenerIntegrationTest() {
 
   @Inject
   private lateinit var queueAdminService: QueueAdminService
 
-  @BeforeEach
-  internal fun setUp() {
-    messageRepository.deleteAll()
```
Comment (mikehalmamoj) [inline src/testIntegration/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/HouseKeepingIntegrationTest.kt:28]: Tidy up between tests has been moved to `IntegrationTest`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt [hunk 3]
```diff
@@ -85,25 +82,4 @@ class JmsConfig {
...
-    @Value("\${sqs.queue.name}") queueName: String,
-    @Value("\${sqs.dlq.name}") dlqName: String
-  ): String {
-    val result = awsSqsClient.createQueue(CreateQueueRequest(dlqName))
-    val dlqArn = awsSqsClient.getQueueAttributes(result.queueUrl, listOf(QueueAttributeName.QueueArn.toString()))
-    awsSqsClient.createQueue(
-      CreateQueueRequest(queueName).withAttributes(
-        mapOf(
-          QueueAttributeName.RedrivePolicy.toString() to
-"""{"deadLetterTargetArn":"${dlqArn.attributes["QueueArn"]}","maxReceiveCount":"5"}"""
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt:103]: Interesting that we need to recreate the queue due to Redrive Policy, why those this no longer matter? (I can't remember why we did it in the first place)

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt (hunk 3 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/JmsConfig.kt:103]: We did it to work around a problem we were having with testcontainers.  I couldn't remember the details either so I removed it to see what broke - nothing broke :man_shrugging:

**Pull request merged**